src/lib/math/bigint/big_ops2.cpp

```diff
@@ -11,6 +11,20 @@
 #include <botan/internal/mp_core.h>
 #include <algorithm>
 
+/* This hack works around an undefined reference to __udivti3() when compiling for 64-bit Windows
+ * using clang-cl, see:
+ *
+ * https://github.com/llvm/llvm-project/issues/25679
+ * https://stackoverflow.com/questions/68676184/clang-cl-error-lld-link-error-undefined-symbol-divti3
+ *
+ * I couldn't come up with a way to embed this into the build info files without extending
+ * configure.py to allow cpu-specific libs in the compiler info or os+cc+arch-specific libs in the
+ * module info. Hopefully this can go away when the above Clang issue is fixed anyway.
+*/
+#if defined(_WIN64) && defined(BOTAN_BUILD_COMPILER_IS_CLANGCL)
+   #pragma comment(lib, "clang_rt.builtins-x86_64.lib")
+#endif
```

Comment:
Is there a 32bit version of this workaround? Not that I think there's a lot of practical use for that in 2024, but perhaps we should try to add a 32bit clangcl build?

Reply:
Apparently that is relevant on 64 bit platforms only.  As per: https://stackoverflow.com/questions/68676184/clang-cl-error-lld-link-error-undefined-symbol-divti3#comment121371051_68676270

> GCC and Clang only support 128-bit integers in 64-bit mode, even in Linux. It's not because of the calling convention because large structs can be still be returned via a hidden pointer

Reply:
Might still be relevant for aarch64 though?

Reply:
Since the issue was opened in 2015 and is still unresolved I'm not holding out a huge amount of hope that it will ever be fixed. So I'd be inclined to try to address this within the build system. I can look at this part.